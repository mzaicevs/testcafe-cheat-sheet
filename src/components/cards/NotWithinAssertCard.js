import React from "react";
import { assert_notWithin } from "../../snippets";
import CodeCard from "../../base_components/CodeCard";

const NotWithinAssertCard = () => (
  <CodeCard
    cardTitle={assert_notWithin.title}
    fullDocLink={assert_notWithin.fullDocLink}
    codeSnippet={{
      code: assert_notWithin.code,
      language: assert_notWithin.language,
      options: assert_notWithin.options,
      example: assert_notWithin.example
    }}
    description={assert_notWithin.description}
  />
);

export default NotWithinAssertCard;
