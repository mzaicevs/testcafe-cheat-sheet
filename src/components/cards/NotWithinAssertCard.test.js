import React from "react";
import { shallow } from "enzyme";
import NotWithinAssertCard from "./NotWithinAssertCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("NotWithinAssertCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<NotWithinAssertCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
