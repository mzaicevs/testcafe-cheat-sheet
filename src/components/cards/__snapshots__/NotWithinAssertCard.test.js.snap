// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotWithinAssertCard matches snapshot 1`] = `
<CodeCard
  cardTitle="Not Within - Assert"
  codeSnippet={
    Object {
      "code": Array [
        "t.expect( actual ).notWithin( start, finish, message, options );",
      ],
      "example": "import { Selector } from 'testcafe';

  fixture \`My fixture\`;
  
  test('My test', async t => {
      await t
        .expect(Selector('#element').scrollTop).notWithin(100, 200)
        .expect(1).notWithin(3, 10, 'this assertion will pass');
  });",
      "language": "javascript",
      "options": "{ 
    timeout: Number,
      allowUnawaitedPromise: Boolean 
  }",
    }
  }
  description="Asserts that actual is not within a range from start to finish. Bounds are inclusive."
  fullDocLink="https://devexpress.github.io/testcafe/documentation/test-api/assertions/assertion-api.html#not-within"
/>
`;
